hive-hooks/path-conversion-hook/src/test/java/com/expediagroup/apiary/extensions/hooks/pathconversion/filters/ApiaryMetastoreFilterTest.java

```diff
@@ -0,0 +1,197 @@
+import org.apache.hadoop.hive.metastore.api.*;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.mockito.stubbing.Answer;
+
+import com.google.common.collect.ImmutableList;
+
+import com.expediagroup.apiary.extensions.hooks.pathconversion.converters.GenericConverter;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ApiaryMetastoreFilterTest {
+    @Mock HiveConf hiveConf;
+    @Mock GenericConverter converter;
+    private MetaStoreFilterHook hook;
+
+    @Before
+    public void init() {
+        Properties mockProps = new Properties();
+        when(hiveConf.getAllProperties()).thenReturn(mockProps);
+        hook = new ApiaryMetastoreFilter(hiveConf, converter);
+    }
+
+    @Test
+    public void shouldNotMutateDatabases() throws MetaException {
+        List<String> dbNames = ImmutableList.of("test");
+        List<String> result = hook.filterDatabases(dbNames);
+        assertEquals(dbNames, result);
+    }
+
+    @Test
+    public void shouldNotMutateDatabase() throws MetaException, NoSuchObjectException {
+        Database db = new Database();
+        Database result = hook.filterDatabase(db);
+        assertEquals(db, result);
+    }
+
+    @Test
+    public void shouldNotMutateTableNames() throws MetaException {
+        List<String> tableNames = ImmutableList.of("test");
+        List<String> result = hook.filterTableNames("db", tableNames);
+        assertEquals(tableNames, result);
+    }
+
+    @Test
+    public void shouldNotMutatePartitionSpec() throws MetaException {
+        List<PartitionSpec> partitionSpecs = ImmutableList.of(new PartitionSpec());
+        List<PartitionSpec> result = hook.filterPartitionSpecs(partitionSpecs);
+        assertEquals(partitionSpecs, result);
+    }
+
+    @Test
+    public void shouldNotMutatePartitionNames() throws MetaException {
+        List<String> partitionNames = ImmutableList.of("test");
+        List<String> result = hook.filterPartitionNames("db", "tbl", partitionNames);
+        assertEquals(partitionNames, result);
+    }
+
+    @Test
+    public void shouldNotMutateIndex() throws MetaException, NoSuchObjectException {
+        Index index = new Index();
+        Index result = hook.filterIndex(index);
+        assertEquals(index, result);
+    }
+
+    @Test
+    public void shouldNotMutateIndexNames() throws MetaException {
+        List<String> indexNames = ImmutableList.of("test");
+        List<String> result = hook.filterIndexNames("db", "tbl", indexNames);
+        assertEquals(indexNames, result);
+    }
+
+    @Test
+    public void shouldNotMutateIndexes() throws MetaException {
+        List<Index> indexList = ImmutableList.of(new Index());
+        List<Index> result = hook.filterIndexes(indexList);
+        assertEquals(indexList, result);
+    }
+
+    @Test
+    public void shouldMutateTables() throws MetaException {
```

Comment:
Your hook doesn't know or manipulate locations your test should reflect that consider this:

```
    @Test
    public void shouldMutateTables() throws MetaException {
        Table sourceTable1 = new Table();
        Table sourceTable2 = new Table();
        List<Table> tableList = ImmutableList.of(sourceTable1,  sourceTable2);
        List<Table> result = hook.filterTables(tableList);
        assertEquals(tableList.size(), result.size());
        verify(converter).convertPath(sourceTable1);
        verify(converter).convertPath(sourceTable2);
    }
```
Which tests exactly the logic of the hook and nothing more. The location conversion testing should be done in the test of the locationConverter.